SCons/Platform/__init__.py

```diff
@@ -192,8 +192,15 @@ def __call__(self, target, source, env, for_signature):
         # Check if we already created the temporary file for this target
         # It should have been previously done by Action.strfunction() call
         node = target[0] if SCons.Util.is_List(target) else target
-        cmdlist = getattr(node.attributes, 'tempfile_cmdlist', None) \
-                    if node is not None else None
+        cmdlist = None
+
+        if isinstance(self.cmd, list):
```

Comment:
`isinstance` rather than `is_List`? (consistency nit, nothing else). Other consistency nit is line 194 uses the `= foo if bar else baz` form and this bit doesn't.  Neither are important to fix, just noting.